### PR TITLE
Colin/electron

### PR DESCRIPTION
**Pull request by uonai**
**To create an electron binary**
$ **npm run dev** 
$ **npm run build**

In server.js (line 27) update to this to ensure that we are building for prod
const app = next({ dev: production, dir: __dirname, quiet: false });

$ npm run electronPack 

macOS binary will now be in the /dist/Mac folder

Click on the application. After application builds (approx 15 - 30 seconds), you need to refresh the electron window. This forces nextJS application to load. 

For debugging the macOS binary, I have been using Debugtron as my primary debugger.

It does appear that the production version of our next.js app has issues with the way it is handling image upload from this endpoint: /_/deals/storage. A request is made to the endpoint but there is no response. Once resolved, this should fix the issue in the electron wrapper.

Note: on first pass there seem to be quite a few updates to the server.js logic since I've started the work on the electron wrapper. None of this was caught with git --rebase. I can manually merge this code if requested.

### CODE REVIEW

**Comment by jimmylee** [comment]
I've done some testing, and on the web version the create wallet address code looks reverted. I'll be taking a look to see where that is happening.

File: node_common/utilities.js [hunk 1]
```diff
@@ -140,9 +140,8 @@ export const refreshLibrary = async ({ state, PG, FFS }) => {
         }
 
         PG.ffs.watchJobs((job) => {
-          // NOTE(jim): FFS is undefined?
           console.log("[ prototype ] job status", job.status);
-          if (job.status >= 5) {
+          if (job.status === FFS.JobStatus.JOB_STATUS_SUCCESS) {
```
Comment (jimmylee) [inline node_common/utilities.js:144]: Please revert this back to whats on the other side. `if (job.status >= 5) {`

File: package.json [hunk 2]
```diff
@@ -8,11 +10,26 @@
     "dev": "node . --unhandled-rejections=strict",
     "build": "next build",
     "start": "NODE_ENV=production node . --unhandled-rejections=strict",
-    "build-system": "rollup -c"
```
Comment (jimmylee) [inline package.json:11]: Please add this back. `"build-system": "rollup -c"`

File: package.json [hunk 3]
```diff
@@ -36,13 +53,17 @@
     "react-dom": "^16.12.0",
     "react-draggable": "^4.4.3",
     "react-tippy": "^1.3.4",
+    "regenerator-runtime": "^0.13.5",
     "three": "^0.108.0",
     "uuid": "^8.0.0",
     "ws": "^7.3.0"
   },
   "devDependencies": {
-    "@rollup/plugin-json": "^4.1.0",
```
Comment (jimmylee) [inline package.json:44]: Please add back this removed dependency.

File: server.js [hunk 4]
```diff
@@ -25,10 +24,11 @@ let state = null;
 const production = process.env.NODE_ENV === "production";
 const port = process.env.PORT || 1337;
 const wsPort = process.env.WS_PORT || 2448;
-const resetData = process.env.npm_config_reset_data;
-const app = next({ dev: !production, quiet: false });
+const app = next({ dev: !production, dir: __dirname, quiet: false });
 const nextRequestHandler = app.getRequestHandler();
 
+const path = require("path");
```
Comment (jimmylee) [inline server.js:30]: Shouldn't this be `import path from 'path';`?

File: server.js [hunk 5]
```diff
@@ -70,103 +70,104 @@ app.prepare().then(async () => {
...
-  if (!production) {
-    try {
-      // NOTE(daniel): Wipe all of the local data when --reset-data flag is added to npm run dev.
-      if (resetData) {
-        await Utilities.resetFileSystem();
-      }
+  try {
+    // TODO(jim): Remove later.
+    // We wipe all of the local data each time you run the application.
+    await Utilities.resetFileSystem();
```
[the remark this answers is not in the document]
Comment (uonai) [reply server.js]: All set w/ latest commit.

File: node_common/utilities.js (hunk 1 above)
Comment (uonai) [reply node_common/utilities.js]: All set w/ latest commit.

File: package.json (hunk 2 above)
Comment (uonai) [reply package.json:11]: All set w/ latest commit.

File: package.json (hunk 3 above)
Comment (uonai) [reply package.json:44]: All set w/ latest commit.

File: server.js (hunk 4 above)
Comment (uonai) [reply server.js]: Thanks for catching this, fixed w/ latest commit.

**Comment by jimmylee** [comment]
LGTM, Landing this now. We'll get to the bottom of the rest of the issues together. Any changes feel free to submit other PRs.

**Pull request merged**